src/packagedcode/go_mod.py

```diff
@@ -73,140 +105,221 @@
-        """
-        Return a dictionary containing all the important go.mod file data.
-        """
-        with io.open(location, encoding='utf-8', closefd=True) as data:
-            lines = data.readlines()
-
-        gomod_data = {}
-        require = []
-        exclude = []
-
-        for i, line in enumerate(lines):
-            line = cls.preprocess(line)
-            parsed_module = cls.parse_module(line)
-            if parsed_module:
-                gomod_data['module'] = parsed_module.group('module')
-
-            parsed_module_name = cls.parse_module_name(line)
-            if parsed_module_name:
-                gomod_data['name'] = parsed_module_name.group('name')
-                gomod_data['namespace'] = parsed_module_name.group('namespace')
-                
-            parsed_require = cls.parse_require(line)
-            if parsed_require:
-                line_req = [parsed_require.group('namespace'), parsed_require.group('name'), parsed_require.group('version')]
-                require.append(line_req)
-
-            parsed_exclude = cls.parse_exclude(line)
-            if parsed_exclude:
-                line_exclude = [parsed_exclude.group('namespace'), parsed_exclude.group('name'), parsed_exclude.group('version')]
-                exclude.append(line_exclude)
-
-            if 'require' in line and '(' in line:
-                for req in lines[i+1:]:
-                    req = cls.preprocess(req)
-                    if ')' in req:
-                        break
-                    parsed_dep_link = cls.parse_dep_link(req)
-                    if parsed_dep_link:
-                        line_req = [parsed_dep_link.group('namespace'), parsed_dep_link.group('name'), parsed_dep_link.group('version')]
-                        require.append(line_req)
-
-            if 'exclude' in line and '(' in line:
-                for exc in lines[i+1:]:
-                    exc = cls.preprocess(exc)
-                    if ')' in exc:
-                        break
-                    parsed_dep_link = cls.parse_dep_link(exc)
-                    if parsed_dep_link:
-                        line_exclude = [parsed_dep_link.group('namespace'), parsed_dep_link.group('name'), parsed_dep_link.group('version')]
-                        exclude.append(line_exclude)
-
-        gomod_data['require'] = require
-        gomod_data['exclude'] = exclude
-
-        return gomod_data
+# Regex expressions to parse different types of go.sum file dependency
+# dep_type1 example: github.com/BurntSushi/toml v0.3.1 h1:WXkYY....
+parse_dep_type1 = re.compile(
+    r'(?P<namespace>(.*))'
+    r'\/'
+    r'(?P<name>[^\s]*)'
+    r'(\s)*'
+    r'(?P<version>(.*))'
+    r'/go.mod(\s)*h1:'
+    r'(?P<checksum>(.*))'
+).match
+
+# dep_type2 example: github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCN....
+parse_dep_type2 = re.compile(
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+    r'(\s)*'
+    r'(?P<version>(.*))'
+    r'(\s)*h1:'
```

Comment:
Since the only difference between type 1 and 2 is this... what about that with a few other refinements:
```Python
get_dependency = re.compile(
    r'(?P<namespace>[^\s]*)'
    r'\/'
    r'(?P<name>[^\s]*)'
    r'\s*'
    r'(?P<version>[^\s]*)'
    r'\s*'
    r'(>P:<is_gomod>/go.mod)?
    r'\s*'
    r'h1:'(?P<checksum>[^\s]*)'
    r'\s*'
).match
```
This may need a few adjustments... but this will also tell you if this is a `go.mod` line (for the checksum, if an when we handle it in the future?)
And by ignoring the spaces everywhere in the capture groups the `.strip()` you used below are no longer needed.

Reply:
I tried this 
```
get_dependency = re.compile(
    r'(?P<namespace>[^\s]*)'
    r'\/'
    r'(?P<name>[^\s]*)'
    r'\s*'
    r'(?P<version>[^\s]*)'
    r'\s*'
    r'(>P:<is_gomod>/go.mod)?'
    r'\s*'
    r'h1:(?P<checksum>[^\s]*)'
    r'\s*'
).match
```
but it was not working. I am suggesting a new change. Maybe this looks good.